src/ext/java/org/opentripplanner/ext/digitransitemissions/EmissionsDataModel.java

```diff
@@ -0,0 +1,55 @@
+package org.opentripplanner.ext.digitransitemissions;
+
+import jakarta.inject.Inject;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+
+public class EmissionsDataModel implements Serializable {
+
+  private Map<FeedScopedId, Emissions> emissions;
+  private double carAvgCo2EmissionsPerMeter;
+
+  @Inject
+  public EmissionsDataModel() {}
+
+  public EmissionsDataModel(
+    Map<FeedScopedId, Emissions> emissions,
+    double carAvgCo2EmissionsPerMeter
+  ) {
+    this.emissions = emissions;
+    this.carAvgCo2EmissionsPerMeter = carAvgCo2EmissionsPerMeter;
+  }
+
+  /**
+   * For testing only.
+   * @param digitransitEmissions
+   */
+  public void setDigitransitEmissions(
+    Map<FeedScopedId, DigitransitEmissions> digitransitEmissions
+  ) {
+    Map<FeedScopedId, Emissions> emissions = new HashMap<>();
+    for (FeedScopedId id : digitransitEmissions.keySet()) {
+      emissions.put(id, digitransitEmissions.get(id));
+    }
+    this.emissions = emissions;
+  }
+
+  public void setEmissions(Map<FeedScopedId, Emissions> emissions) {
+    this.emissions = emissions;
+  }
+
+  public Optional<Map<FeedScopedId, Emissions>> getEmissions() {
```

Comment:
While not a hard rule, I think we are settling on not using the word "get" in getter methods anymore.

Reply:
Also, isn't a `Map` already an optional type? What's the difference between an empty `Optional` and an `Optional` with an empty `Map` in it?

If there is none, you can return `Map.of()` instead of an Optional.

Reply:
> While not a hard rule, I think we are settling on not using the word "get" in getter methods anymore. Just call it emissions().

This is not correct, see https://github.com/opentripplanner/OpenTripPlanner/blob/dev-2.x/CODE_CONVENTIONS.md. This is not a DTO/VO/Entity, but a Module/Service type, so use proper verbs for retrieving data. We do not use `get`/`set` on Java "Bean". There is a convention for what `get` and `find` means. Also passing out the field here is breaking encapsulation.

[This is inside the Sandbox, so I have not read the code, I just happen to read @leonardehrenfried  comment]

Reply:
This is refactored now.